applications/nrf5340_audio/src/modules/sd_card_playback.c

```diff
@@ -0,0 +1,426 @@
+		if (ret < 0) {
+			LOG_ERR("SD card read err: %d", ret);
+			sd_card_playback_active = false;
+			return ret;
+		}
+		if (pcm_wav_mono_frame_size < pcm_frame_size) {
+			sd_card_playback_active = false;
+		}
+		/* Wait until there is enough space in the ringbuffer */
+		if (sd_card_playback_active) {
+
+			ret = sd_card_playback_ringbuf_write(pcm_wav_mono_frame,
+							     pcm_wav_mono_frame_size);
+			if (ret < 0) {
+				LOG_ERR("Load ringbuf err: %d.", ret);
+				sd_card_playback_active = false;
+				return ret;
+			}
+		}
+	}
+
+	ret = sd_card_close(&f_seg_read_entry);
+	if (ret < 0) {
+		LOG_ERR("SD card close file err: %d", ret);
+		return ret;
+	}
+
+	return 0;
+}
+
+static void sd_card_playback_thread(void *arg1, void *arg2, void *arg3)
+{
+	int ret;
+	while (1) {
+		if (!sw_codec_is_initialized()) {
+			k_msleep(200);
+			continue;
+		}
+		k_sem_take(&m_sem_playback, K_FOREVER);
+		switch (playback_file_format) {
+		case SD_CARD_PLAYBACK_WAV:
+			ret = sd_card_playback_play_wav();
+			if (ret < 0) {
+				LOG_ERR("Wav playback err: %d", ret);
+			}
+			break;
+		case SD_CARD_PLAYBACK_LC3:
+			ret = sd_card_playback_play_lc3();
+			if (ret < 0) {
+				LOG_ERR("LC3 playback err: %d", ret);
+			}
+			break;
+		}
+	}
+}
+
+bool sd_card_playback_is_active(void)
+{
+	return sd_card_playback_active;
+}
+
+void sd_card_playback_wav(char *filename, uint32_t frame_duration_ms, uint8_t bit_depth,
+			  enum sd_playback_sample_rates sample_rate)
+{
+	playback_file_format = SD_CARD_PLAYBACK_WAV;
+	playback_file_name = filename;
+	wav_playback_cfg.frame_duration_ms = frame_duration_ms;
+	wav_playback_cfg.bit_depth = bit_depth;
+	wav_playback_cfg.sample_rate = sample_rate;
+	k_sem_give(&m_sem_playback);
+}
+
+int sd_card_playback_lc3(char *filename)
+{
+	if (!sw_codec_is_initialized()) {
+		LOG_ERR("Sw codec not initialized");
+		return -EACCES;
+	}
+	playback_file_format = SD_CARD_PLAYBACK_LC3;
+	playback_file_name = filename;
+	k_sem_give(&m_sem_playback);
+	return 0;
+}
+
+int sd_card_playback_mix_with_stream(void *const pcm_a, size_t pcm_a_size)
+{
+	int ret;
+	uint8_t pcm_b[pcm_frame_size];
+	size_t read_size = pcm_frame_size;
+
+	ret = sd_card_playback_ringbuf_read(pcm_b, &read_size);
+	if (ret < 0) {
+		LOG_ERR("Loading data into buffer err: %d", ret);
+		return ret;
+	}
+	if (read_size == pcm_frame_size) {
+		ret = pcm_mix(pcm_a, pcm_a_size, pcm_b, pcm_frame_size, B_MONO_INTO_A_STEREO_L);
+		if (ret < 0) {
+			LOG_ERR("Pcm mix err: %d", ret);
+		}
+	} else {
+		LOG_WRN("Size written (%d) does not equal requested size (%d). Skipping...",
+			read_size, pcm_frame_size);
+	}
+	return 0;
+}
+
+int sd_card_playback_init(void)
+{
+	int ret;
+
+	sd_card_playback_thread_id = k_thread_create(
+		&sd_card_playback_thread_data, sd_card_playback_thread_stack,
+		CONFIG_SD_CARD_PLAYBACK_STACK_SIZE, (k_thread_entry_t)sd_card_playback_thread, NULL,
+		NULL, NULL, CONFIG_SD_CARD_PLAYBACK_THREAD_PRIORITY, 0, K_NO_WAIT);
```

Comment:
To stay in sync with other names: 
```suggestion
		NULL, NULL, K_PRIO_PREEMPT(CONFIG_SD_CARD_PLAYBACK_THREAD_PRIO), 0, K_NO_WAIT);
```